src/start_menu/main_settings/change_text_weight.cpp

```diff
@@ -4,55 +4,51 @@
 
 void ChangeTextWeight()
 {
-	do
-	{
-		std::cout << "Pick Weight" << std::endl;
-		std::cout << "1. light" << std::endl;
-		std::cout << "2. SemiLight" << std::endl;
-		std::cout << "3. Normal" << std::endl;
-		std::cout << "4. SemiBold" << std::endl;
-		std::cout << "5. Bold" << std::endl;
-
-		int weight_shoice;
-		utils::GetUserInput(weight_shoice);
+	std::cout << "Pick text weight:" << std::endl;
+	std::cout << "1. Light" << std::endl;
+	std::cout << "2. SemiLight" << std::endl;
+	std::cout << "3. Normal" << std::endl;
+	std::cout << "4. SemiBold" << std::endl;
+	std::cout << "5. Bold" << std::endl;
 
-		switch (weight_shoice)
-		{
-			case 1:
-				{
-					utils::g_text_weight = utils::FontWeightValues::Light;
-					break;
-				}
-			case 2:
-				{
-					utils::g_text_weight = utils::FontWeightValues::SemiLight;
-					break;
-				}
-			case 3:
-				{
-					utils::g_text_weight = utils::FontWeightValues::Normal;
-					break;
-				}
-			case 4:
-				{
-					utils::g_text_weight = utils::FontWeightValues::SemiBold;
-					break;
-				}
-			case 5:
-				{
-					utils::g_text_weight = utils::FontWeightValues::Bold;
-					break;
-				}
-			default:
-				{
-					std::cout << "Invalid choice! Please choose again." << std::endl;
-					continue;
-				}
-		}
+	int text_weight_choice = 0;
+	utils::GetUserInput(text_weight_choice);
 
-		utils::CustomizeText(); // Make the new changes take effect
+	while (text_weight_choice < 1 || text_weight_choice > 5)
+	{
+		std::cout << "Sorry, this option is not supported yet.\nPlease choose "
+					 "one of the options provided: ";
+		utils::GetUserInput(text_weight_choice);
+	}
```

Comment:
Is it wise to add a loop to the code? it only makes the complexity of the code worse, this goes for the rest of the changes as well.

Reply:
I am not sure what complexity for this type of while loop even means. The flag for the `while` statement will run at least once, and then as many times as the user enters a wrong input, together with the two statements inside the code block. Otherwise, the block will be skipped.

As far as I know, there isn't any other way to handle incorrect user input, where you want to repeatedly ask the user for a correct input before the program moves on.